### PR TITLE
feat: Optional ID parameter in page templates

### PR DESCRIPTION
**Pull request by broksy**
Added optional `ID` property to use in page manifest templates.

### CODE REVIEW

**Review by schreckstefan — DISMISSED** [review]
The enhancement makes sense,  is compatible, and covered by tests

**Review by 815are — COMMENTED** [review]
`id` is consumed for object page and list report, but it is not cosumed for custom page ->
https://github.com/SAP/open-ux-tools/blob/feat/pageId/packages/fe-fpm-writer/templates/page/custom/1.94/manifest.json#L5

is it expected?

File: packages/fe-fpm-writer/test/unit/page/__snapshots__/custom.test.ts.snap [hunk 1]
```diff
@@ -807,6 +851,48 @@ exports[`CustomPage generateCustomPage: different versions or target folder with
...
+        },
+        Object {
+          "name": "DummyPage",
+          "pattern": "RootEntity:?query:",
+          "target": "DummyPage",
+        },
+      ],
+      "targets": Object {
+        "DummyPage": Object {
+          "options": Object {
```
Comment (schreckstefan) [inline packages/fe-fpm-writer/test/unit/page/__snapshots__/custom.test.ts.snap:880]: id is missing here. Is this an error?

Comment (broksy) [reply packages/fe-fpm-writer/test/unit/page/__snapshots__/custom.test.ts.snap:880]: Test is for 1.84 manifest - https://github.com/SAP/open-ux-tools/blob/main/packages/fe-fpm-writer/templates/page/custom/1.84/manifest.json template, it doesn't use `id` property.

**Review by schreckstefan — APPROVED** [review]
Tested OK, creation of custom pages works as before on root and subordinate level.

**Review by 815are — CHANGES_REQUESTED** [review]
Following looks missed:
1. Object page template - https://github.com/SAP/open-ux-tools/blob/feat/pageId/packages/fe-fpm-writer/templates/page/object/manifest.json#L20
`route` in source page still has old `<%- entity %>ObjectPage` even if `id` is passed
2. Custom page template - https://github.com/SAP/open-ux-tools/blob/feat/pageId/packages/fe-fpm-writer/templates/page/custom/1.94/manifest.json#L20
https://github.com/SAP/open-ux-tools/blob/feat/pageId/packages/fe-fpm-writer/templates/page/custom/1.84/manifest.json#L19
Same as in 1st point.
3. Please add tests where you add new object page and custom page with custom 'id' and with valid source page. Current snapshots does not cover it -> https://github.com/SAP/open-ux-tools/blob/feat/pageId/packages/fe-fpm-writer/test/unit/page/__snapshots__/object.test.ts.snap#L193-L194

**Review by 815are — APPROVED** [review]
checked:
1. code changes enables feature to provide custom id for new object, list, custom pages;
2. new tests are added with and without source page;
3. change log is present

did not run manual tests

**Review by schreckstefan — APPROVED** [review]

**Pull request merged**